scripts/clear-all-faults.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# This shell script sets the group D-Bus objects in
+# /xyz/openbmc_project/State//Decorator/OperationalStatusManager
+# to true or false. Also Remove the critical association from
+# motherboard/chassis to Inventory D-Bus object on which it sets
+# the OperationalStatus.
+
+function usage()
+{
+    echo "clear-all-faults.sh [true/false]"
+    echo "Example: clear-all-faults.sh true"
+    echo "Example: clear-all-faults.sh false bmc_booted power_on _fault"
+    return 0;
+}
+
+# We need at least 1 argument
+if [ $# -lt 1 ]; then
+    echo "At least ONE argument needed";
+    usage;
+    exit 1;
+fi
+
+# User passed in argument [true/false]
+action=$1
+
+# If it is not "true" or "false", exit
+if [ "$action" != "true" ] && [ "$action" != "false" ]; then
+    echo "Bad argument $action passed";
+    usage;
+    exit 1;
+fi
+
+# Get the excluded groups, where $@ is all the agruments passed
+index=2;
+excluded_groups=""
+
+for arg in "$@"
+do
+   if [ "$arg" == "$action" ]
+   then
+       # Must be true/false
+       continue
+   elif [ $index -eq $# ]
```

Comment:
A comment would be good here, this logic isn't all that obvious to me.

Reply:
Ack